### ISSUE TITLE
set_defaults blocks displaying of dataframe data (widget and column names are displayed).

### ISSUE THREAD

**Issue by BjarkeDalslet** [quantopian/qgrid#157]
Calling set_defaults with grid_options != None causes the widget to be displayed without any data in the columns - only the header (column names and filter buttons) are shown.
Subsequent calls of qgrid.set_defaults(grid_options = None) does not remove the problem.

### Environment

* Operating System: Windows 7
* Python Version: '3.6.3 |Anaconda custom (64-bit)| (default, Oct 15 2017, 03:27:45) [MSC v.1900 64 bit (AMD64)]'
* How did you install Qgrid: Conda using the instructions on the github page verbatim. No errors. Did conda update conda and conda update anaconda first without problems.
* Python packages: # packages in environment at C:\ProgramData\Anaconda3:
#
_ipyw_jlab_nb_ext_conf    0.1.0            py36he6757f0_0  
alabaster                 0.7.10           py36hcd07829_0  
anaconda                  custom           py36h363777c_0  
anaconda-client           1.6.5            py36hd36550c_0  
anaconda-navigator        1.6.9            py36hc720852_0  
anaconda-project          0.8.0            py36h8b3bf89_0  
asn1crypto                0.22.0           py36h8e79faa_1  
astroid                   1.5.3            py36h9d85297_0  
astropy                   2.0.2            py36h06391c4_4  
babel                     2.5.0            py36h35444c1_0  
backports                 1.0              py36h81696a8_1  
backports.shutil_get_terminal_size 1.0.0            py36h79ab834_2  
beautifulsoup4            4.6.0            py36hd4cc5e8_1  
bitarray                  0.8.1            py36h6af124b_0  
bkcharts                  0.2              py36h7e685f7_0  
blaze                     0.11.3           py36h8a29ca5_0  
bleach                    2.0.0            py36h0a7e3d6_0  
bokeh                     0.12.10          py36h0be3b39_0  
boto                      2.48.0           py36h1a776d2_1  
bottleneck                1.2.1            py36hd119dfa_0  
bzip2                     1.0.6                    vc14_1  [vc14]  conda-forge
ca-certificates           2017.08.26           h94faf87_0  
cachecontrol              0.12.3           py36hfe50d7b_0  
certifi                   2017.7.27.1      py36h043bc9e_0  
cffi                      1.10.0           py36hae3d1b5_1  
chardet                   3.0.4            py36h420ce6e_1  
click                     6.7              py36hec8c647_0  
cloudpickle               0.4.0            py36h639d8dc_0  
clyent                    1.2.2            py36hb10d595_1  
colorama                  0.3.9            py36h029ae33_0  
comtypes                  1.1.2            py36heb9b3d1_0  
conda                     4.3.29                   py36_0    conda-forge
conda-build               3.0.22           py36hd71664c_0  
conda-env                 2.6.0                         0    conda-forge
conda-verify              2.0.0            py36h065de53_0  
console_shortcut          0.1.1                h6bb2dd7_3  
contextlib2               0.5.5            py36he5d52c0_0  
cryptography              2.0.3            py36h123decb_1  
curl                      7.55.1                   vc14_0  [vc14]  conda-forge
cycler                    0.10.0           py36h009560c_0  
cython                    0.26.1           py36h18049ac_0  
cytoolz                   0.8.2            py36h547e66e_0  
dask                      0.15.3           py36h396fcb9_0  
dask-core                 0.15.3           py36hd651449_0  
datashape                 0.5.4            py36h5770b85_0  
decorator                 4.1.2            py36he63a57b_0  
dfgui                     0.1                      py36_0    file:///C:/Temp/PandasDataFrameGUI-master/dfgui
dill                      0.2.6                    py36_0    conda-forge
distlib                   0.2.5            py36h51371be_0  
distributed               1.19.1           py36h8504682_0  
docutils                  0.14             py36h6012d8f_0  
entrypoints               0.2.3            py36hfd66bb0_2  
et_xmlfile                1.0.1            py36h3d2d736_0  
fastcache                 1.0.2            py36hffdae1b_0  
filelock                  2.0.12           py36hd7ddd41_0  
flask                     0.12.2           py36h98b5e8f_0  
flask-cors                3.0.3            py36h8a3855d_0  
freetype                  2.8.1                    vc14_0  [vc14]  conda-forge
get_terminal_size         1.0.0                h38e98db_0  
gevent                    1.2.2            py36h342a76c_0  
glob2                     0.5              py36h11cc1bd_1  
graphviz                  2.38.0                        5    conda-forge
greenlet                  0.4.12           py36ha00ad21_0  
gtabview                  0.8                       <pip>
h5py                      2.7.0            py36hfbe0a52_1  
hdf5                      1.10.1                   vc14_1  [vc14]  conda-forge
heapdict                  1.0.0            py36h21fa5f4_0  
html5lib                  0.999999999      py36ha09b1f3_0  
icc_rt                    2017.0.4             h97af966_0  
icu                       58.2                     vc14_0  [vc14]  conda-forge
idna                      2.6              py36h148d497_1  
imageio                   2.2.0            py36had6c2d2_0  
imagesize                 0.7.1            py36he29f638_0  
intel-openmp              2018.0.0             hcd89f80_7  
ipykernel                 4.6.1            py36hbb77b34_0  
ipython                   6.1.0            py36h236ecc8_1  
ipython_genutils          0.2.0            py36h3c5d0ee_0  
ipywidgets                7.0.0            py36h2e74ada_0  
isort                     4.2.15           py36h6198cc5_0  
itsdangerous              0.24             py36hb6c5a24_1  
jdcal                     1.3              py36h64a5255_0  
jedi                      0.10.2           py36hed927a0_0  
jinja2                    2.9.6            py36h10aa3a0_1  
jpeg                      9b                       vc14_2  [vc14]  conda-forge
jsonschema                2.6.0            py36h7636477_0  
jupyter                   1.0.0            py36h422fd7e_2  
jupyter_client            5.1.0            py36h9902a9a_0  
jupyter_console           5.2.0            py36h6d89b47_1  
jupyter_contrib_core      0.3.3                    py36_0    conda-forge
jupyter_contrib_nbextensions 0.3.1                    py36_0    conda-forge
jupyter_core              4.3.0            py36h511e818_0  
jupyter_highlight_selected_word 0.0.10                   py36_0    conda-forge
jupyter_latex_envs        1.3.8.2                  py36_1    conda-forge
jupyter_nbextensions_configurator 0.2.8                    py36_0    conda-forge
jupyterlab                0.27.0           py36h34cc53b_2  
jupyterlab_launcher       0.4.0            py36h22c3ccf_0  
krb5                      1.14.2                   vc14_0  [vc14]  conda-forge
lazy-object-proxy         1.3.1            py36hd1c21d2_0  
libiconv                  1.14                     vc14_4  [vc14]  conda-forge
libpng                    1.6.34                   vc14_0  [vc14]  conda-forge
libssh2                   1.8.0                    vc14_2  [vc14]  conda-forge
libtiff                   4.0.9                    vc14_0  [vc14]  conda-forge
libxml2                   2.9.5                    vc14_1  [vc14]  conda-forge
libxslt                   1.1.32                   vc14_0  [vc14]  conda-forge
line_profiler             2.0                      py36_0    conda-forge
llvmlite                  0.20.0                   py36_0  
locket                    0.2.0            py36hfed976d_1  
lockfile                  0.12.2           py36h0468280_0  
lxml                      4.1.0            py36h0dcd83c_0  
lzo                       2.10                     vc14_0  [vc14]  conda-forge
markupsafe                1.0              py36h0e26971_1  
matplotlib                2.1.0            py36h11b4b9c_0  
mccabe                    0.6.1            py36hb41005a_1  
menuinst                  1.4.10           py36h42196fb_0  
mistune                   0.7.4            py36h4874169_0  
mkl                       2018.0.0             h36b65af_4  
mkl-service               1.1.2            py36h57e144c_4  
mpmath                    0.19             py36he326802_2  
msgpack-python            0.4.8            py36h58b1e9d_0  
multipledispatch          0.4.9            py36he44c36e_0  
navigator-updater         0.1.0            py36h8a7b86b_0  
nbconvert                 5.3.1            py36h8dc0fde_0  
nbformat                  4.4.0            py36h3a5bc1b_0  
networkx                  2.0              py36hff991e3_0  
nltk                      3.2.4            py36hd0e0a39_0  
nose                      1.3.7            py36h1c3779e_2  
notebook                  5.0.0            py36hd9fbf6f_2  
numba                     0.35.0             np113py36_10  
numexpr                   2.6.2            py36h7ca04dc_1  
numpy                     1.13.3           py36ha320f96_0  
numpydoc                  0.7.0            py36ha25429e_0  
odo                       0.5.1            py36h7560279_0  
olefile                   0.44             py36h0a7bdd2_0  
openpyxl                  2.4.8            py36hf3b77f6_1  
openssl                   1.0.2n                   vc14_0  [vc14]  conda-forge
packaging                 16.8             py36ha0986f6_1  
pandas                    0.20.3           py36hce827b7_2  
pandoc                    1.19.2.1             hb2460c7_1  
pandocfilters             1.4.2            py36h3ef6317_1  
partd                     0.3.8            py36hc8e763b_0  
path.py                   10.3.1           py36h3dd8b46_0  
pathlib2                  2.3.0            py36h7bfb78b_0  
patsy                     0.4.1            py36h42cefec_0  
pep8                      1.7.0            py36h0f3d67a_0  
pickleshare               0.7.4            py36h9de030f_0  
pillow                    4.2.1            py36hdb25ab2_0  
pip                       9.0.1            py36hadba87b_3  
pkginfo                   1.4.1            py36hb0f9cfa_1  
ply                       3.10             py36h1211beb_0  
pox                       0.2.3                    py36_0    conda-forge
progress                  1.3              py36hbeca8d3_0  
prompt_toolkit            1.0.15           py36h60b8f86_0  
psutil                    5.4.0            py36h4e662fb_0  
py                        1.4.34           py36ha4aca3a_1  
pycodestyle               2.3.1            py36h7cc55cd_0  
pycosat                   0.6.2            py36hf17546d_1  
pycparser                 2.18             py36hd053e01_1  
pycrypto                  2.6.1            py36he68e6e2_1  
pycurl                    7.43.0           py36h086bf4c_3  
pyflakes                  1.6.0            py36h0b975d6_0  
pygments                  2.2.0            py36hb010967_0  
pylint                    1.7.4            py36ha4e6ded_0  
pyodbc                    4.0.17           py36h0006bc2_0  
pyopenssl                 17.2.0           py36h15ca2fc_0  
pyparsing                 2.2.0            py36h785a196_1  
pypyodbc                  1.3.4                     <pip>
pyqt                      5.6.0            py36hb5ed885_5  
pyreadline                2.1                      py36_0  
pysocks                   1.6.7            py36h698d350_1  
pytables                  3.4.2            py36h71138e3_2  
pytest                    3.2.1            py36h753b05e_1  
python                    3.6.3                h9e2ca53_1  
python-dateutil           2.6.1            py36h509ddcb_1  
python-graphviz           0.8                      py36_0    conda-forge
pytz                      2017.2           py36h05d413f_1  
pywavelets                0.5.2            py36hc649158_0  
pywin32                   221              py36h9c10281_0  
pyyaml                    3.12             py36h1d1928f_1  
pyzmq                     16.0.2           py36h38c27d9_2  
qgrid                     1.0.0b10                 py36_0    tim_shawver/label/dev
qt                        5.6.2                    vc14_1  [vc14]  conda-forge
qtawesome                 0.4.4            py36h5aa48f6_0  
qtconsole                 4.3.1            py36h99a29a9_0  
qtpy                      1.3.1            py36hb8717c5_0  
requests                  2.18.4           py36h4371aae_1  
rope                      0.10.5           py36hcaf5641_0  
ruamel_yaml               0.11.14          py36h9b16331_2  
scikit-image              0.13.0           py36h6dffa3f_1  
scikit-learn              0.19.1           py36h53aea1b_0  
scipy                     0.19.1           py36h7565378_3  
seaborn                   0.8.0            py36h62cb67c_0  
setuptools                36.5.0           py36h65f9e6e_0  
simplegeneric             0.8.1            py36heab741f_0  
singledispatch            3.4.0.3          py36h17d0c80_0  
sip                       4.18.1           py36h9c25514_2  
six                       1.11.0           py36h4db2310_1  
snowballstemmer           1.2.1            py36h763602f_0  
sortedcollections         0.5.3            py36hbefa0ab_0  
sortedcontainers          1.5.7            py36ha90ac20_0  
sphinx                    1.6.3            py36h9bb690b_0  
sphinxcontrib             1.0              py36hbbac3d2_1  
sphinxcontrib-websupport  1.0.1            py36hb5e5916_1  
spyder                    3.2.4            py36h8845eaa_0  
sqlalchemy                1.1.13           py36h5948d12_0  
sqlite                    3.20.1                   vc14_2  [vc14]  conda-forge
statsmodels               0.8.0            py36h6189b4c_0  
sympy                     1.1.1            py36h96708e0_0  
tabview                   1.4.2                     <pip>
tblib                     1.3.2            py36h30f5020_0  
testpath                  0.3.1            py36h2698cfe_0  
tk                        8.6.7                    vc14_0  [vc14]  conda-forge
toolz                     0.8.2            py36he152a52_0  
tornado                   4.5.2            py36h57f6048_0  
traitlets                 4.3.2            py36h096827d_0  
typing                    3.6.2            py36hb035bda_0  
unicodecsv                0.14.1           py36h6450c06_0  
urllib3                   1.22             py36h276f60a_0  
vc                        14                   h2379b0c_2  
vs2015_runtime            14.0.25123           hd4c4e62_2  
wcwidth                   0.1.7            py36h3d5aa90_0  
webencodings              0.5.1            py36h67c50ae_1  
werkzeug                  0.12.2           py36h866a736_0  
wheel                     0.29.0           py36h6ce6cde_1  
widgetsnbextension        3.0.2            py36h364476f_1  
win_inet_pton             1.0.1            py36he67d7fd_1  
win_unicode_console       0.5              py36hcdbd4b5_0  
wincertstore              0.2              py36h7fe50ca_0  
wrapt                     1.10.11          py36he5f5981_0  
wxpython                  4.0.0a3                  py36_0    conda-forge
xlrd                      1.1.0            py36h1cb58dc_1  
xlsxwriter                1.0.2            py36hf723b7d_0  
xlwings                   0.11.4           py36hd3cf94d_0  
xlwt                      1.3.0            py36h1a4751e_0  
yaml                      0.1.7                    vc14_0  [vc14]  conda-forge
zict                      0.1.3            py36h2d8e73e_0  
zlib                      1.2.11                   vc14_0  [vc14]  conda-forge

### Reproduction Steps
All done in the "vanilla" jupyter notebook 

This only displays the column names (and filter buttons) in the widget:
```
import pandas
df=pandas.DataFrame({"mobs":[1,2,3,4]})
import qgrid
qgrid.set_defaults(grid_options={
            'fullWidthRows': True,
            'syncColumnCellResize': True,
            'forceFitColumns': True,
            'rowHeight': 28,
            'enableColumnReorder': False,
            'enableTextSelectionOnCells': True,
            'editable': True,
            'autoEdit': False
        })
qgrid.show_grid(df)
```

----

This displays the column data as it should be displayed:
```
import pandas
df=pandas.DataFrame({"mobs":[1,2,3,4]})
import qgrid
qgrid.set_defaults(grid_options=None)
qgrid.show_grid(df)
```

**Comment by BjarkeDalslet** [comment]
Addendum / workaround:
Setting default options individually using e.g.  `qgrid.set_grid_option("forceFitColumns", False)`
does _not_ block the displaying of dataframe data .

**Comment by Etherian** [comment]
I also experience this issue.
- Windows 7
- Anaconda 64-bit
- Python 3.6
- Firefox 57

(That's all the context I can think of right now. I'll add the rest next time I'm at my computer.)

Note: The grid is rendered properly in the full-screen modal. Opening and closing the modal causes the inline widget to show the missing cells.

**Comment by TimShawver** [comment]
I forgot to update part of the API docs to add some new default options for grid_options in version 1.0.  Sorry about the confusion...I've committed the change to the docs on master but I'm having some trouble getting the read-the-docs page updated so the API docs are not up-to-date as of writing.  

Here's the new default dictionary for grid_options that you'll see in the API docs, once they're updated:
```
{
            'fullWidthRows': True,
            'syncColumnCellResize': True,
            'forceFitColumns': True,
            'defaultColumnWidth': 150,
            'rowHeight': 28,
            'enableColumnReorder': False,
            'enableTextSelectionOnCells': True,
            'editable': True,
            'autoEdit': False,
            'explicitInitialization': True,
            'maxVisibleRows': 15,
            'minVisibleRows': 8
        }
```

The last two options `maxVisibleRows` and `minVisibleRows` are options that I added specifically for qgrid.  In other words you won't see them in the SlickGrid documentation.  If you include those in your options dictionary, the grid should be sized normally.  At some point I'll improve the default behavior so we don't end up with a zero height grid.
